rwengine/src/data/GameTexts.hpp

```diff
@@ -3,21 +3,42 @@
 #include <string>
 #include <unordered_map>
 
+/**
+ * The game stores strings as 16-bit indexes into the font
+ * texture, which is something simllar to ASCII.
+ */
+using GameString = std::basic_string<char16_t>;
```

Comment:
(See my other comment below, I didn't see we already used 2 different types)

What I meant with the second part was: For our custom strings, your intention was to use unicode strings (u"blah") which assigns a meaning to the symbols. We only display those strings with the game font map, so we can't assign meanings to game characters unless we make assumptions about the fontmap.
(You assigned that meaning in the u"" prefixed strings and moved that meaning to a `char16_t` (GameString = there shouldn't be a meaning here unless we know the fontmap))

Let's assume that the fontmap is ASCII for 'A'-'Z'. We could store our custom strings in `std::string` and do the conversion with `fromString` (only supporting 'A'-'Z'). Alternatively we could store a `GameString` in the source code directly, as it would appear in the GXT file (meaning as an array, maybe with a comment describing what it represents in a specific fontmap).

So, there is no point in prefixing all strings with u"". The fact that custom strings and game strings have the same width doesn't matter. Unless the host compiler couldn't express the subset we want to map (such as 'A'-'Z') in `char` / `std::string` OR we want to map unicode symbols for a japanese font map or something (from our custom strings).